src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -62,34 +72,50 @@ public bool IsStarted
         /// <param name="elapsed">The task to run.</param>
         public void Start(Func<Task> elapsed)
         {
-            var newTokenSource = new CancellationTokenSource();
+            DelayedWork delayedWork = new DelayedWork(elapsed);
+            DelayedWork previousWork = Interlocked.Exchange(ref this.latestWork, delayedWork);
+
+            if (previousWork != null)
+            {
+                previousWork.IsCancellationRequested = true;
```

Comment:
When I am thinking of respective scenarios, I imagine maybe a dozen of additional timers. This should not matter. Can you think of scenarios where the impact is bigger. If so, please clarify and I may need to rethink my approach.
@olegsych